doc/design/wallet/wallet-design.md

```diff
@@ -0,0 +1,768 @@
+///   Encrypted tags can be searched only for exact matching
+///   Note if some from provided tags already assigned to the record than
+///     corresponding tags values will be replaced
+extern pub fn indy_add_wallet_record_tags(command_handle: i32,
+                                          wallet_handle: i32,
+                                          type_: *const c_char,
+                                          id: *const c_char,
+                                          tags_json: *const c_char,
+                                          cb: Option<extern fn(command_handle_: i32, err: ErrorCode)>) -> ErrorCode {}
+
+/// Delete tags from the wallet record
+///
+/// #Params
+/// command_handle: command handle to map callback to caller context
+/// wallet_handle: wallet handle (created by open_wallet)
+/// type_: allows to separate different record types collections
+/// id: the id of record
+/// tag_names_json: the list of tag names to remove from the record as json array:
+///   ["tagName1", "tagName2", ...]
+///   Note that null means no tag names
+extern pub fn indy_delete_wallet_record_tags(command_handle: i32,
+                                             wallet_handle: i32,
+                                             type_: *const c_char,
+                                             id: *const c_char,
+                                             tag_names_json: *const c_char,
+                                             cb: Option<extern fn(command_handle_: i32, err: ErrorCode)>) -> ErrorCode {}
+
+/// Delete an existing wallet record in the wallet
+///
+/// #Params
+/// command_handle: command handle to map callback to caller context
+/// wallet_handle: wallet handle (created by open_wallet)
+/// type_: record type
+/// id: the id of record
+extern pub fn indy_delete_wallet_record(command_handle: i32,
+                                        wallet_handle: i32,
+                                        type_: *const c_char,
+                                        id: *const c_char,
+                                        cb: Option<extern fn(command_handle_: i32, err: ErrorCode)>) -> ErrorCode> {}
+
+/// Get an wallet record by id
+///
+/// #Params
+/// command_handle: command handle to map callback to caller context
+/// wallet_handle: wallet handle (created by open_wallet)
+/// type_: allows to separate different record types collections
+/// id: the id of record
+/// options_json: //TODO: FIXME: Think about replacing by bitmaks
+///  {
+///    retrieveValue: (optional, true by default) Retrieve record value,
+///    retrieveTags: (optional, true by default) Retrieve record tags
+///  }
+/// #Returns
+/// wallet record json:
+/// {
+///   id: "Some id",
+///   value: "Some value", // present only if retrieveValue set to true
+///   tags: <tags json>, // present only if retrieveTags set to true
+/// }
+extern pub fn indy_get_wallet_record(command_handle: u32,
+                                     wallet_handle: i32,
```

Comment:
handles are sometimes i32, sometimes u32, which one is correct?

Reply:
Correct, i32 for handles. Count should be usize. I will update.